File: TypeScript/0704-二分查找.ts
function search(nums: number[], target: number): number {
	let left: number = 0,
		right: number = nums.length - 1;
	while (left <= right) {
		const mid = left + ((right - left) >> 1);
		const num: number = nums[mid];
		if (num < target) {
			left = mid + 1;
		} else if (num > target) {
			right = mid - 1;
		} else {
			return mid;
		}
	}
	return -1;
}

File: Java/0704-二分查找.java
class Solution {
  public int search(int[] nums, int target) {
    if (target < nums[0] || target > nums[nums.length - 1]) {
      return -1;
    }
    int left = 0, right = nums.length - 1, mid = 0, num = 0;
    while (left <= right) {
      mid = left + ((right - left) >> 1);
      num = nums[mid];
      if (num == target) {
        return mid;
      } else if (num < target) {
        left = mid + 1;
      } else {
        right = mid - 1;
      }
    }
    return -1;
  }
}
